### DISCUSSION TITLE
Aesthetic Privileges (And Badges)

### DISCUSSION THREAD

**Discussion by Kacarott** [codewars/codewars.com#2369] [Ideas]
Simple idea, but the privilege progression track could add some more 'appealing' (to some users) rewards at the higher levels of Honor. Some examples:

- User can set name color (Maybe within a limited palette to avoid eye-bleed)
- User can set a (or multiple?) badge next to their name. (Specific examples of a badge would be language icons, to show off preferred/proficient languages)
    - Regarding badges, this could almost be a seperate idea itself. They could replace the mysterious "Achievements" which (to my knowledge) aren't actually visible anywhere. Most would be milestones (reaching 2/1 kyu in a language for a silver/gold badge of that lang icon, badges for reaching 2, 1 kyu, 1, 2, (3?) dan, badges for completing some number of 1kyu katas, badges for being above 6kyu in some large number of langs, etc). But maybe some badges are also given as one-time events during "special events"
- Potentially some kind of customisation of profile page? (Show off your badges!)

Some of this might clash with, or be better for "Red" users, not sure. I am also not sure how hard all this is to implement/whether it is worth it, but I thought I would at least post the idea.

**Comment by ejini6969** [comment]
Maybe badges for authoring (only for approved katas) or translating (approved ones as well) certain amount of katas can be made too to encourage more people participating in content-making.

**Comment by hobovsky** [comment]
I would love to have awards for solving problems and for authoring activities completely separated. Users could be, for example, awarded with honor when completing a kata, and they would receive some badge or some kind of brownie points for creating a kata, translation, upvotes of all kinds, etc.

In my opinion, current way of mixing honor for solving and for authoring does not really make sense, and all effects of collecting "overall honor", like system of privilege rewards or the "overall" leaderboard, do not accurately reflect anything.

Or maybe just get rid of honor completely, and just leave rank awarded for problems, and "recognized author/translator" badge(s)/points for authoring?

**Comment by kazk** [comment]
My comment on achievements and Codewars gamification in general.

> > While I agree that ranks used on Codewars have various issues, I don't think adding few dan level will fix anything. I wish we had more flexible and systematic ways to describe the difficulty. **Trying to fit challenges from different areas and different languages into a single scale doesn't work.**
> > 
> > I also don't think _"could discourage Codewarriors from authoring and/or solving the more difficult ones"_ will be an issue. I don't think most of us are doing for honor points. We simply like to solve/create interesting problems.  
> > [#1769 (comment)](https://github.com/codewars/codewars.com/issues/1769#issuecomment-473726359)
> 
> Codewars' gamification is broken like most of the other things. Introducing new ranks won't fix anything. It needs fundamental changes, reorganizing and reranking, but this requires too much work for us (me) to handle at the moment.
> 
> More realistic approach might be to come up with something to complement the current system like achievement badges. If "Codewars Battleground" (experimental invite-only competition for Codewars Red users taken on Qualified), is successful, (I think) we're going to introduce some kind of trophy/badge so we might get time to work on it.  
> Achievement badges are more suitable to handle special cases like this.
> 
> Maybe we can create [Codewars/badges](https://github.com/Codewars/badges) to maintain the list of badges. Allow admins and hand picked users to assign badges to kata (issues on that repo can be used to suggest/discuss). Then add a section on the profile for badges. We'll have to keep it simple at first, but this might be possible with relatively low effort.
> 
> What do you think? Badges are also great to show detailed abilities like `Coq: Proved {insert something to be proud of}` which I think is way more meaningful than `solved 1 dan Coq kata` where `1 dan` is assessed by _few users_ for _a difficulty within Codewars_.
>
> https://github.com/codewars/codewars.com/issues/1859#issuecomment-506263320

**Comment by kazk** [comment]
[Introduce achievements/badges · Issue #604 · codewars/codewars.com](https://github.com/codewars/codewars.com/issues/604)

> StackOverflow.com has badges, video-games often have achievements, and I believe Code-Wars would benefit from adding badges & achievements as well. It should be tangent-to/separate-from the honor system (i.e., it is not based on honor-points).
> 
> Here are some ideas:
> 
> > **\[Nice Solution\]** : A badge you can get by having a solution that earns 3+ Best-Practices votes  
> > **\[Clever Devil\]** : (Similar to the above) but for Clever solutions  
> > **\[Revisionist\]** : You get by editing/correcting other katas  
> > **\[Burgeoning Author\]** : (one-time) You get for authoring your first kata  
> > **\[Prolific Author\]** : (one-time) You get by authoring lots of kata  
> > **\[Silver Solution\]** : Getting 10 Best-Practice votes  
> > **\[Gold Solution\]** : Getting 20+ Best-Practice votes  
> > **\[Poly-Glot\]** : Getting above a 5-kyu in 3+ languages  
> > etc... The Medal modifier (i.e., "Gold/Silver" could be extended to create ranks of badges, "Bronze/Silver/Gold/Platinum _Whatever Badge_".
> 
> It would be nice for a badge to be a link in the user's profile to the thing that earned the badge (i.e., the solution, comment, translation, etc.).
